index.js

```diff
@@ -0,0 +1,213 @@
+  }
+}
+
+// delta = b*b - 4*a*c
+const getDelta = getDeltaFunction({ a: 1, b: 2, c: 3 });
+console.log(getDelta()); // -8
+console.log(getDelta({ b: 12 })); // 132
+console.log(getDelta({ a: 4, b: 10 })); // 52
+console.log(getDelta({ a: 4, b: 0 })); // -48
+
+// The nested function returns NaN if any of the arguments are missing
+console.log(getDeltaFunction({ a: 1, c: 3 })()); // NaN
+console.log(getDeltaFunction({ a: 1, c: 3 })({ b: 15 })); // 213
+
+// 2
+function positiveSum(numbers) {
+  let numbersSum = 0;
+
+  numbers.forEach(function(number) {
+    if (number > 0) {
+      numbersSum = numbersSum + number;
+    }
+  })
+
+  return numbersSum;
+}
+
+console.log(positiveSum([1, 5, 10]));
+console.log(positiveSum([-1, 5, -10]));
+
+// 3
+function countSheep(shed) {
+  return shed.filter(Boolean).length;
+}
+
+const shed = [true, false, true, false, true];
+console.log(countSheep(shed)); //3
+
+// 4
+function findNeedle(haystack) {
+  const needle = (el) => el === "needle";
+  const index = haystack.findIndex(needle);
+  return 'found the needle at position ' + index;
+}
+
+const haystack = ["hay", "junk", "hay", "hay", "moreJunk", "needle", "randomJunk"];
+console.log(findNeedle(haystack));
+
+// 5
+function doubleNumbers(numbers){
+  return numbers.map(function(number) {
+    return number * 2;
+  })
+}
+
+console.log(doubleNumbers([1, 2, 3]));
+
+// 6
+function invertNumbers(numbers) {
+  return numbers.map(function(number) {
+    return number * -1;
+  })
+}
+
+console.log(invertNumbers([1, -2, 3, -4, 5]));
+
+// 7
+function getAverage(marks){
+  let averageMarks = 0;
+
+  marks.forEach(function(mark) {
+    averageMarks += mark;
+  })
+
+  return Math.floor(averageMarks / marks.length);
+}
+
+console.log(getAverage([3, 4, 5]));
+
+// 8
+function isStringInArray(array) {
+  return !!array.find(function(string) {
+    return typeof string === 'string';
+  })
+}
+
+console.log(isStringInArray([4, 'Orange'], 'Apple')); // true
+console.log(isStringInArray([3, 4], 'Potato')); // false
+
+// 9
+function forEach(array, callback) {
+  for (let i = 0; i < array.length; ++i) {
+    callback(array[i], i.toString());
+  }
+}
+
+const vegetables = ['Carrot', 'Cabbage', 'Onion'];
+
+function printVegetable(vegetable, index) {
+  console.log(vegetable, index);
+}
+
+forEach(vegetables, printVegetable);
+// Carrot 0
+// Cabbage 1
+// Onion 2
+
+const fruits = ['Apple', 'Orange', 'Watermelon'];
+forEach(fruits,
+  function(fruit, index) {
+    console.log(fruit, index);
+  }
+);
+// Apple 0
+// Orange 1
+// Watermelon 2
+
+// 10
+function executeAfterFiveSeconds(callback) {
+  setTimeout(() => {
+    callback();
+  }, 5000);
+}
+
+function sayHello() {
+  console.log('Hello!');
+}
+
+executeAfterFiveSeconds(sayHello);
+
+// 11
+function getRandomIntegerGenerator(firstNumber, secondNumber) {
+  const subtraction = (secondNumber - firstNumber);
+  return Math.floor(Math.random() * subtraction) + firstNumber;
```

Comment:
This is still too complicated. What `Math.random() * subtraction` means? Create variables with descriptive names.